src/librustc_mir/interpret/eval_context.rs

```diff
@@ -1583,7 +1583,7 @@ impl<'a, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M
                                 let alloc_kind = self.tcx.alloc_map.lock().get(ptr.alloc_id);
                                 if let Some(AllocType::Static(did)) = alloc_kind {
                                     // extern statics should not be validated
```

Comment:
This should maybe say "cross-crate" or "non-local", not to be confused with `extern { static }`.

Reply:
oh this is about extern statics, and non-local statics

Reply:
I'm confused, you're not checking `is_foreign_item`.

Reply:
`has_typeck_tables` :smiling_imp:

Reply:
But that's just... that's not... please don't use that method. If you are checking for foreign statics, i.e. `extern { static }`, use `is_foreign_item` for clarity, if nothing else.